prow/metrics/prowjobs/collector_test.go

```diff
@@ -0,0 +1,191 @@
+package prowjobs
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+)
+
+func TestProwJobLifecycleCollectorUpdate(t *testing.T) {
+	fakeClock := clock.NewFakeClock(time.Now())
+	defaultCompleteTime := v1.NewTime(fakeClock.Now().Add(1 * time.Hour))
+	defaultPendingTime := v1.NewTime(fakeClock.Now().Add(30 * time.Minute))
+	type args struct {
+		oldJob *prowapi.ProwJob
+		newJob *prowapi.ProwJob
+	}
+	type expected struct {
+		collected []dto.Metric
+	}
+	tests := []struct {
+		oldJobStates []prowapi.ProwJobState
+		newJobStates []prowapi.ProwJobState
+		name         string
+		args         args
+		expected     expected
+	}{
+		{name: "should not collect job without history when no state change was observed for state %s to %s",
+			oldJobStates: []prowapi.ProwJobState{prowapi.TriggeredState, prowapi.PendingState, prowapi.SuccessState, prowapi.FailureState, prowapi.ErrorState, prowapi.AbortedState},
+			newJobStates: []prowapi.ProwJobState{prowapi.TriggeredState, prowapi.PendingState, prowapi.SuccessState, prowapi.FailureState, prowapi.ErrorState, prowapi.AbortedState},
+			args: args{
+				oldJob: &prowapi.ProwJob{
+					ObjectMeta: v1.ObjectMeta{
+						UID:               "1234",
+						CreationTimestamp: v1.NewTime(fakeClock.Now()),
+					},
+					Status: prowapi.ProwJobStatus{
+						CompletionTime: &defaultCompleteTime,
+						PendingTime:    &defaultPendingTime,
+					},
+				},
+				newJob: &prowapi.ProwJob{
+					ObjectMeta: v1.ObjectMeta{
+						UID:               "1234",
+						CreationTimestamp: v1.NewTime(fakeClock.Now()),
+					},
+					Status: prowapi.ProwJobStatus{
+						CompletionTime: &defaultCompleteTime,
+						PendingTime:    &defaultPendingTime,
+					},
+				},
+			}, expected: expected{
+				collected: nil,
+			}},
+		{name: "should collect job transitions for trasitions from %s to  %s",
```

Comment:
we misspell `transitions` a number of different ways in this file

Reply:
Fixed.